### ISSUE TITLE
UnicodeDecodeError when calling detect_language()

### ISSUE THREAD

**Issue by PetersonZhao** [trezor/python-mnemonic#51]
In `detect_language()`, at line 81:
```python
for lang in languages:
    mnemo = cls(lang)
```
This will call` __init__()`:
```python
def __init__(self, language):
    self.radix = 2048
    with open('%s/%s.txt' % (self._get_directory(), language), 'r') as f:
        self.wordlist = [w.strip().decode('utf8') if sys.version < '3' else w.strip() for w in f.readlines()]
    if len(self.wordlist) != self.radix:
        raise ConfigurationError('Wordlist should contain %d words, but it contains %d words.' % (self.radix, len(self.wordlist)))
```
And open "japanese.txt" with encoding "US-ASCII", and then decode error raised:
```
/Users/peter.zhao/.pyenv/versions/py36-blockchain/lib/python3.6/site-packages/mnemonic/mnemonic.py:146: in to_mnemonic
    if self.detect_language(' '.join(result)) == 'japanese':  # Japanese must be joined by ideographic space.
/Users/peter.zhao/.pyenv/versions/py36-blockchain/lib/python3.6/site-packages/mnemonic/mnemonic.py:81: in detect_language
    mnemo = cls(lang)
/Users/peter.zhao/.pyenv/versions/py36-blockchain/lib/python3.6/site-packages/mnemonic/mnemonic.py:51: in __init__
    self.wordlist = [w.strip() for w in f.readlines()]
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <encodings.ascii.IncrementalDecoder object at 0x10d9dc6a0>
input = b'\xe3\x81\x82\xe3\x81\x84\xe3\x81\x93\xe3\x81\x8f\xe3\x81\x97\xe3\x82\x93\n\xe3\x81\x82\xe3\x81\x84\xe3\x81\x95\xe3\x...93\xe3\x81\x86\xe3\x81\xaa\xe3\x81\x84\n\xe3\x81\x93\xe3\x81\x86\xe3\x81\xaf\xe3\x81\x84\n\xe3\x81\x93\xe3\x82\x99\xe3'
final = False

    def decode(self, input, final=False):
>       return codecs.ascii_decode(input, self.errors)[0]
E       UnicodeDecodeError: 'ascii' codec can't decode byte 0xe3 in position 0: ordinal not in range(128)
```

I edit the code to this: 
```python
def __init__(self, language):
    self.radix = 2048
    with open('%s/%s.txt' % (self._get_directory(), language), 'r', encoding='utf8') as f:
        self.wordlist = [w.strip().decode('utf8') if sys.version < '3' else w.strip() for w in f.readlines()]
    if len(self.wordlist) != self.radix:
        raise ConfigurationError('Wordlist should contain %d words, but it contains %d words.' % (self.radix, len(self.wordlist)))
```
and now everything is ok in my environment. But I'm not sure this will work fine in other versions of Python, like, 2.7.

**Comment by jameshilliard** [comment]
This bug seems to be due to `detect_language()` getting called even in cases when a language was explicitly passed to the init method, `detect_language()` should be refactored to always use the language specified in `def __init__(self, language)`.

I ended up overriding `detect_language()` like this:
```python
from mnemonic import Mnemonic as MnemonicParent
class Mnemonic(MnemonicParent):
    @classmethod
    def detect_language(cls, code):
        return "english"
```

> It looks like your locale is configured incorrectly.

Local shouldn't matter if you explicitly pass the language to the `Mnemonic` instance so this seems like a library bug.

**Comment by jameshilliard** [comment]
Another option may be just to remove this call [here](https://github.com/trezor/python-mnemonic/blob/8e628d8f546a9db2272f9c559bd2a94133d473c2/mnemonic/mnemonic.py#L146) and use the language passed to `init` instead. I don't see the purpose of trying to detect the language when that should already be known.

**Comment by matejcik** [comment]
this has been fixed since c78bdbe6ac0a855b048dcbf13ef79e5145b6181c
